src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py

```diff
@@ -1006,8 +1007,16 @@ def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code:
         logger.warning(f"Found existing {flow_yaml_path.as_posix()}, will not respect it in runtime.")
         with open(flow_yaml_path, "r", encoding=DEFAULT_ENCODING) as f:
             existing_content = f.read()
-    if not is_local_module(entry_string=entry, code=code):
-        logger.debug(f"Entry {entry} is not found in local, it's snapshot will be empty.")
+
+    create_yaml_in_temp_dir = False
+    if is_eval_batch_run():
```

Comment:
From the architecture's view, I think devkit should not understand "is_eval_batch_run"?
Can we use a more clear signal to hint these?

Reply:
Understood. How about changing it to create_flow_entry_in_temp() with the environment variable PF_CREATE_FLOW_ENTRY_IN_TEMP? Would that work? Let me know if you have other suggestions